source/slang/slang-serialize-types.h

```diff
@@ -0,0 +1,232 @@
+// slang-serialize-types.h
+#ifndef SLANG_SERIALIZE_TYPES_H
+#define SLANG_SERIALIZE_TYPES_H
+
+#include "../core/slang-riff.h"
+#include "../core/slang-string-slice-pool.h"
+#include "../core/slang-array-view.h"
+
+#include "slang-name.h"
+#include "slang-source-loc.h"
+
+namespace Slang {
+
+// Options for IR/AST/Debug serialization
+
+struct SerialOptionFlag
+{
+    typedef uint32_t Type;
+    enum Enum : Type
+    {
+        RawSourceLocation   = 0x01,     ///< If set will store directly SourceLoc - only useful if current source locs will be identical when read in (typically this is *NOT* the case)
```

Comment:
It seems like we are basically at the point where we need/want to drop `RawSourceLocation` support, since our long-term goals for serialization are almost all incompatible with it.